### PR TITLE
UI/create login page

### LINKED ISSUE #6: feat: create login page
Creation of the login page, which is displayed on the first startup of the app when there is no cache.

On this page, users are asked to enter their INE, date of birth and name. 

No verification is performed on the fields at present (in future, the INE and date of birth will be used to authenticate on the 3IL student area).

All these fields are cached (sharedPreferences), so you don't have to request them again every time you start the app.

### PR DESCRIPTION
**Pull request by PHPLukaas**
Close #6 
- Form data stored in sharedPreferences
- Redirect user to schedule if already logged in + selected class
- Class selection after login

### CODE REVIEW

File: .github/workflows/deploy-backend-dev.yml [hunk 1]
```diff
@@ -4,9 +4,6 @@ on:
   push:
     branches:
       - dev
-  pull_request:
-    branches:
```
Comment (julesartd) [inline .github/workflows/deploy-backend-dev.yml:8]: question: what is the reason for deleting this code?

File: frontend/app_student/lib/login/cubit/login_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    String? name = prefs.getString('name');
+    String? birthDate = prefs.getString('birthDate');
+    String? className = prefs.getString('className');
+
+    if (ine != null && name != null && birthDate != null && className != null) {
+      emit(LoginAuthenticated());
+    } else {
+      emit(LoginInitial());
+    }
+  }
```
Comment (julesartd) [inline frontend/app_student/lib/login/cubit/login_cubit.dart:45]: suggestion: update the birthdate to a DateTime maybe?
but that's cool nice job!

File: frontend/app_student/lib/login/widgets/form/form_login.dart [hunk 3]
```diff
@@ -5,18 +5,31 @@ import 'inputs/input_birthdate.dart';
...
+    return Scaffold(
       body: Column(
         children: [
-          INETextField(),
-          BirthDateField(),
-          FirstnameTextField(),
-          SubmitButton()
+          INETextField(controller: ineController),
+          BirthDateField(controller: birthDateController),
+          FirstnameTextField(controller: nameController),
```
Comment (julesartd) [inline frontend/app_student/lib/login/widgets/form/form_login.dart:27]: suggestion: encompass form fields instead of displaying them 1 by 1

**Review by julesartd — APPROVED** [review]
it's good overall, I accept the pull request but there will be some small details to modify

File: .github/workflows/deploy-backend-dev.yml (hunk 1 above)
Comment (PHPLukaas) [reply .github/workflows/deploy-backend-dev.yml:8]: Deployment in dev was launched during a PR even in draft so I only put the push on the branch.

**Pull request merged**